patches/api/0008-Paper-Plugins.patch

```diff
@@ -0,0 +1,2486 @@
++    // Paper end
++        this.loader = new io.papermc.paper.plugin.provider.util.DummyBukkitPluginLoader(this);
+         this.server = server;
+         this.file = file;
+         this.description = description;
+@@ -278,6 +298,7 @@ public abstract class JavaPlugin extends PluginBase {
+         this.classLoader = classLoader;
+         this.configFile = new File(dataFolder, "config.yml");
+         this.logger = new PluginLogger(this);
++        this.pluginMeta = configuration; // Paper
+     }
+ 
+     /**
+diff --git a/src/main/java/org/bukkit/plugin/java/JavaPluginLoader.java b/src/main/java/org/bukkit/plugin/java/JavaPluginLoader.java
+index 047c0304fd617cec990f80815b43916c6ef5a94c..d0ad072c832b8fc8a1cfdcafdd42c724531a2e29 100644
+--- a/src/main/java/org/bukkit/plugin/java/JavaPluginLoader.java
++++ b/src/main/java/org/bukkit/plugin/java/JavaPluginLoader.java
+@@ -49,6 +49,7 @@ import org.yaml.snakeyaml.error.YAMLException;
+ /**
+  * Represents a Java plugin loader, allowing plugins in the form of .jar
+  */
++@Deprecated(forRemoval = true) // Paper - The PluginLoader system will not function in the near future. This implementation will be moved.
+ public final class JavaPluginLoader implements PluginLoader {
+     final Server server;
+     private final Pattern[] fileFilters = new Pattern[]{Pattern.compile("\\.jar$")};
+@@ -79,6 +80,7 @@ public final class JavaPluginLoader implements PluginLoader {
+     @Override
+     @NotNull
+     public Plugin loadPlugin(@NotNull final File file) throws InvalidPluginException {
++        if (true) throw new UnsupportedOperationException(); // Paper
+         Preconditions.checkArgument(file != null, "File cannot be null");
+ 
+         if (!file.exists()) {
+@@ -142,7 +144,7 @@ public final class JavaPluginLoader implements PluginLoader {
+ 
+         final PluginClassLoader loader;
+         try {
+-            loader = new PluginClassLoader(this, getClass().getClassLoader(), description, dataFolder, file, (libraryLoader != null) ? libraryLoader.createLoader(description) : null);
++            loader = new PluginClassLoader(getClass().getClassLoader(), description, dataFolder, file, (libraryLoader != null) ? libraryLoader.createLoader(description) : null, null); // Paper
+         } catch (InvalidPluginException ex) {
+             throw ex;
+         } catch (Throwable ex) {
+diff --git a/src/main/java/org/bukkit/plugin/java/LibraryLoader.java b/src/main/java/org/bukkit/plugin/java/LibraryLoader.java
+index 6d634b0ea813ccb19f1562a7d0e5a59cea4eab21..4eec9316e4447701732c944fa84c64a009a5c394 100644
+--- a/src/main/java/org/bukkit/plugin/java/LibraryLoader.java
++++ b/src/main/java/org/bukkit/plugin/java/LibraryLoader.java
+@@ -36,7 +36,7 @@ import org.eclipse.aether.transport.http.HttpTransporterFactory;
+ import org.jetbrains.annotations.NotNull;
+ import org.jetbrains.annotations.Nullable;
+ 
+-class LibraryLoader
++public class LibraryLoader
```

Comment:
Add the `Internal` annotation to declare this as Not API.

Reply:
Yeah I don't think this needs to be public anyways. I think this was made public for when I was originally adding the library loader a LONG time ago.

Reply:
Nevermind.